<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>deep_q_networkClass</class>
 <widget class="QWidget" name="deep_q_networkClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>767</width>
    <height>865</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>deep_q_network</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_5">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,10,0,0">
     <item>
      <widget class="QSlider" name="horizontalSlider">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTableWidget" name="tableWidget"/>
     </item>
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_13" stretch="0">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_17">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="0,10">
             <item>
              <widget class="QLabel" name="label_5">
               <property name="text">
                <string>陷阱数:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_trap">
               <property name="minimum">
                <number>1</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="0,10">
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>终点数:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_target">
               <property name="minimum">
                <number>1</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_5">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,10">
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>状态行:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_row">
               <property name="minimum">
                <number>2</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,10">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>状态列:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_col">
               <property name="minimum">
                <number>2</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_make_state">
             <property name="text">
              <string>生成状态</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_4" stretch="10,0">
     <item>
      <widget class="QWidget" name="widget_loss" native="true"/>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_20" stretch="0,10">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,10">
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>隐藏层神经元个数:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_neural_num">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>200</number>
               </property>
               <property name="value">
                <number>100</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="0,10">
             <item>
              <widget class="QLabel" name="label_8">
               <property name="text">
                <string>学习率:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="doubleSpinBox_learning_rate">
               <property name="decimals">
                <number>5</number>
               </property>
               <property name="value">
                <double>0.005000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_18" stretch="5,5">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="0,10">
             <item>
              <widget class="QLabel" name="label_4">
               <property name="text">
                <string>折扣:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="doubleSpinBox_gama">
               <property name="decimals">
                <number>3</number>
               </property>
               <property name="minimum">
                <double>0.001000000000000</double>
               </property>
               <property name="maximum">
                <double>9.999000000000001</double>
               </property>
               <property name="value">
                <double>0.900000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_9" stretch="0,10">
             <item>
              <widget class="QLabel" name="label_7">
               <property name="text">
                <string>误差:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="doubleSpinBox_error">
               <property name="decimals">
                <number>5</number>
               </property>
               <property name="minimum">
                <double>0.000000000000000</double>
               </property>
               <property name="maximum">
                <double>10.000000000000000</double>
               </property>
               <property name="value">
                <double>0.005000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_19">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_16">
             <item>
              <widget class="QLabel" name="label_10">
               <property name="text">
                <string>轨迹长度:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_step_count">
               <property name="minimum">
                <number>2</number>
               </property>
               <property name="maximum">
                <number>100000000</number>
               </property>
               <property name="value">
                <number>1000</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_14" stretch="0,10">
             <item>
              <widget class="QLabel" name="label_9">
               <property name="text">
                <string>多久更新一次模型:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_model_num">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
               <property name="value">
                <number>500</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_15">
             <item>
              <widget class="QLabel" name="label_11">
               <property name="text">
                <string>经验回放次数:</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinBox_replay_num">
               <property name="minimum">
                <number>1</number>
               </property>
               <property name="maximum">
                <number>10000</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_12">
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_dqn">
             <property name="text">
              <string>DQN</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="pushButton_qlearning">
             <property name="text">
              <string>Qlearning</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_4">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="horizontalSpacer_6">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="deep_q_network.qrc"/>
 </resources>
 <connections/>
</ui>
